MapboxCoreNavigation/RouteController.swift

```diff
@@ -976,6 +977,15 @@ extension RouteController: CLLocationManagerDelegate {
             }
         }
     }
+    
+    func updateVisualInstructionProgress(for location: CLLocation) {
+        let currentStepProgress = routeProgress.currentLegProgress.currentStepProgress
+        guard let visualInstructions = currentStepProgress.remainingVisualInstructions else { return }
+        
+        let visualInstructionDistances = visualInstructions.map { $0.distanceAlongStep }
+        currentStepProgress.visualInstructionIndex = visualInstructionDistances.index { currentStepProgress.distanceRemaining <= $0 } ?? visualInstructions.startIndex
```

Comment:
Pull this out into a variable do only update the visualInstructionIndex if this found variable is not equal to visualInstructionIndex.